driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/BatchCursorPublisherGeneric.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.reactivestreams.client;
+
+import com.mongodb.lang.Nullable;
+import org.reactivestreams.Publisher;
+
+/**
+ * Operations that allow asynchronous iteration over a collection view.
+ *
+ * @param <C> the batch cursor type
+ * @param <T> the result type
+ * @since 4.2
+ */
+public interface BatchCursorPublisherGeneric<C, T> extends Publisher<T> {
```

Comment:
I think it would be clearer if the generic types were

     <C extends BatchCursor<T>, T>
     
It also seems that this whole interface could be lifted into BatchCursorPublisher

Reply:
Updated this base generic interface